src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -36,16 +36,19 @@ internal sealed partial class PowerShellAssemblyLoadContext
         /// <summary>
         /// Initialize a singleton of PowerShellAssemblyLoadContext.
         /// </summary>
-        internal static PowerShellAssemblyLoadContext InitializeSingleton(string basePaths)
+        internal static PowerShellAssemblyLoadContext InitializeSingleton(string basePaths, bool throwOnReentry)
```

Comment:
I feel that the very fact of the appearance of a new parameter indicates that tricky logic is preserved in the startup code. Why not make this initialization more predictable, simple and linear?

Reply:
I don't understand. The existing logic is to not allow this method to be called multiple times from native code.